### ISSUE TITLE
As someone approving pull requests, I want automation to move them in to QA like before

### ISSUE THREAD

**Issue by pdurbin** [IQSS/dataverse#8388]
The main repo has been suffering from a problem for months where pull requests are not automatically moved into QA like before.

As far as I know, this works fine for all other repos besides the main "dataverse" repo.

I believe Danny already tried deleting and recreating the automation but we could try this again.

We can also open a support ticket with GitHub.

Finally, we could try their new forum: https://github.com/github/feedback/discussions

**Comment by pdurbin** [comment]
Try creating a new QA column?

Estimate is about time boxing. How much time to spend.

**Comment by pdurbin** [comment]
As described in the message below that I just sent to GitHub support, I tried disabling automation on the old "QA" column and creating a new column called "QA2" with the same automation ("Approved by reviewer"), but the problem persists.

The message I sent is below:

We've been using automation for years to move pull requests to a column called "QA" at https://github.com/orgs/IQSS/projects/2

A few months ago, this stopped working.

I tried disabling automation on the old "QA" column and creating a new column called "QA2" with the same automation ("Approved by reviewer"), but the problem persists. When I click Approve, the pull request doesn't move. https://github.com/IQSS/dataverse/pull/8413 and https://github.com/IQSS/dataverse/pull/8378 are examples of pull requests that didn't move from "Review" to "QA2" after I clicked "Approve".

I don't see any way I can fix this. Perhaps there is something on the backend you can look at? Please help. Thanks!

Phil

![Screen Shot 2022-02-15 at 3 57 49 PM](https://user-images.githubusercontent.com/21006/154148024-3a420f1f-4578-4348-9a82-6dae8af54410.png)

**Comment by pdurbin** [comment]
Related discussion:

- https://github.community/t/project-board-to-do-automation-has-stopped-working/3297/14
- https://github.com/github/feedback/discussions/6056

**Comment by pdurbin** [comment]
As I mentioned in Slack the other day, I deleted the original QA column and renamed QA2 to QA. This didn't help. Clicking "Approve" still doesn't move pull requests to QA.

I still haven't heard from GitHub support.

I did just asked for help in the new feedback forum at GitHub: https://github.com/github/feedback/discussions/11728

I'll also give an update at standup but I feel like I've put a "small" worth of effort into this so we should probably take it off the board. We can pick it up again if we hear from GitHub or the forum.

**Comment by pdurbin** [comment]
I just looked into the timing of when the automation stopped working and the change at least correlates with when we set up branch protection rules in #8166.

- Nov 22: Automation worked to put a PR into QA: #8253
- Nov 23: I set up branch protection rules: https://github.com/IQSS/dataverse/issues/8166#issuecomment-977062110
- Nov 29: A PR was moved manually to QA, presumable because automation had stopped working: #8257

As a test we could try disabling branch protection temporarily. If automation works again, we can try to find a branch protection configuration that's compatible with automation.

**Comment by pdurbin** [comment]
Good news! It turns out that adding a branch protection rule indeed caused the "move to QA" automation ("Approved by reviewer") to stop working.

I adjusted the rule for `develop` (but not `master`) by checking the box for "Require approvals" and setting the required number to "1" like this:

<img width="797" alt="Screen Shot 2022-03-03 at 1 46 16 PM" src="https://user-images.githubusercontent.com/21006/156639305-0a03d8a6-5c37-4daa-afd1-a1f56019f2af.png">

Then we tested with pull request #8458 and it worked! Hitting "Approve" on the PR moved it to QA. Phew.

I should note that while the rule says approvals are required before merging, this is only true for non-admins. When admins see "merging is blocked" like this...

<img width="930" alt="Screen Shot 2022-03-03 at 2 37 28 PM" src="https://user-images.githubusercontent.com/21006/156639934-ee03ff25-58dd-49f2-bf50-004539f69dcc.png">

... they can click "Merge pull request" anyway and then check the "Use your administrator privileges to merge this pull request" box and merge anyway, like this:

<img width="609" alt="Screen Shot 2022-03-03 at 2 14 55 PM" src="https://user-images.githubusercontent.com/21006/156639307-a0efb6b3-db50-408a-ae73-2ac26c9b49db.png">

So it's an extra step for admins but I think that's fine. Overall, the branch protection rules are nice. Here's a screenshot of how the `develop` rules look. Again, the only change is to check the box requiring approvals and setting it to 1, so it's only a slight change from what we first set up in #8166.

<img width="533" alt="Screen Shot 2022-03-03 at 2 31 58 PM" src="https://user-images.githubusercontent.com/21006/156639309-4bcf2e2b-385c-4e87-ba5a-9a12eb547f61.png">

Moving to review. No code to look at. Just this comment, really.
